File: src/Application Specific/Classes/Entity Collections/EngineerLevel/EngineerLevel.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace EngineerLevels

Public Class EngineerLevel

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "EngineerLevel Properties"

      
Private _EngineerLevelID As Integer = 0
Public Readonly Property EngineerLevelID() As Integer
	Get 
		Return _EngineerLevelID
	End Get	
End Property	
Public Writeonly Property SetEngineerLevelID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_EngineerLevelID", Value)
	End Set
End Property


Private _LevelID As Integer = 0
Public Readonly Property LevelID() As Integer
	Get 
		Return _LevelID
	End Get	
End Property	
Public Writeonly Property SetLevelID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_LevelID", Value)
	End Set
End Property


Private _EngineerID As Integer = 0
Public Readonly Property EngineerID() As Integer
	Get 
		Return _EngineerID
	End Get	
End Property	
Public Writeonly Property SetEngineerID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_EngineerID", Value)
	End Set
End Property


  
#End Region

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/EngineerLevel/EngineerLevel.cs
using System.Collections;

namespace FSM.Entity
{
    namespace EngineerLevels
    {
        public class EngineerLevel
        {
            private DataTypeValidator _dataTypeValidator;

            public EngineerLevel()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _EngineerLevelID = 0;

            public int EngineerLevelID
            {
                get
                {
                    return _EngineerLevelID;
                }
            }

            public object SetEngineerLevelID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EngineerLevelID", value);
                }
            }

            private int _LevelID = 0;

            public int LevelID
            {
                get
                {
                    return _LevelID;
                }
            }

            public object SetLevelID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_LevelID", value);
                }
            }

            private int _EngineerID = 0;

            public int EngineerID
            {
                get
                {
                    return _EngineerID;
                }
            }

            public object SetEngineerID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EngineerID", value);
                }
            }



            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}